src/apps/HeyGPT/HeyGPT/Services/ActivationService.cs

```diff
@@ -1,60 +1,168 @@
+/// and advanced activation scenarios such as handling URL protocols.
+/// </summary>
 public class ActivationService : IActivationService
 {
-    private readonly ActivationHandler<LaunchActivatedEventArgs> _defaultHandler;
+    private readonly IActivationHandler _defaultHandler;
     private readonly IEnumerable<IActivationHandler> _activationHandlers;
     private readonly IThemeSelectorService _themeSelectorService;
-    private UIElement? _shell = null;
+    private UIElement? _shell;
+    private int _activationCount;
 
-    public ActivationService(ActivationHandler<LaunchActivatedEventArgs> defaultHandler, IEnumerable<IActivationHandler> activationHandlers, IThemeSelectorService themeSelectorService)
+    //public ActivationService(ActivationHandler<AppActivationArguments> defaultHandler, IEnumerable<IActivationHandler> activationHandlers, IThemeSelectorService themeSelectorService)
+    public ActivationService(
+        [FromKeyedServices(ActivationHandlerKeys.DefaultServiceKey)] IActivationHandler defaultHandler,
+        [FromKeyedServices(ActivationHandlerKeys.OtherServiceKey)] IEnumerable<IActivationHandler> activationHandlers,
+        IThemeSelectorService themeSelectorService)
     {
         _defaultHandler = defaultHandler;
         _activationHandlers = activationHandlers;
         _themeSelectorService = themeSelectorService;
     }
 
-    public async Task ActivateAsync(object activationArgs)
+    public async Task LaunchAsync(LaunchActivatedEventArgs _)
     {
+        // The application is configured to be a singled instance application,
+        // which means that we need to handle two types of activation scenarios.
+        // When the application is first launched and subsequent activation events
+        // such as receiving a URL protocol (e.g. for OAuth redirection). In
+        // WindowsAppSDK, you handle the initial launch in the App.OnLaunched override
+        // method and secondary activation events through the AppInstance "Activated"
+        // event. This method is called from the App.OnLaunched method, which is only
+        // called once. 
+        //
+        // This method will fetch the current AppInstance and subscribe to its Activated
+        // event. It will then process the first launch scenario, which will process the
+        // main window and initial application startup.  It will also forward the activation
+        // event to any additional handlers that are registered.
+        //
+        // The "Activated" event handler will simply forward the latest activation event
+        // to the other interested parties.
+
+        var currentInstance = AppInstance.GetCurrent();
+
+        if (currentInstance is not null && currentInstance.IsCurrent)
+        {
+            currentInstance.Activated += async (_, arguments) =>
+            {
+                await HandleSubsequentActivationAsync(arguments).ConfigureAwait(false);
```

Comment:
All activation requests that happen AFTER the initial launch will arrive into the application here.